spyder_kernels/utils/iofuncs.py

```diff
@@ -288,58 +289,101 @@ def save_dictionary(data, filename):
     old_cwd = getcwd()
     os.chdir(osp.dirname(filename))
     error_message = None
+    skipped_keys = []
+    data_copy = {}
 
-    # Copy dictionary before modifying to fix #6689
     try:
-        data = copy.deepcopy(data)
-    except NotImplementedError:
-        try:
-            data = copy.copy(data)
-        except Exception:
-            data = data
+        # Copy dictionary before modifying it to fix #6689
```

Comment:
If this PR fixes that Spyder issue, please update the description with the corresponding info.

The good thing about Github is that it can close issues among repos.

Reply:
That comment was there before; it just got moved around with its accompanying code block. I already fixed that issue (although this PR implements a better actual fix). Thanks for checking.